File: Snowballs.cpp
#include <iostream>
#include <cmath>  // Include the cmath library for pow function

double calculateSnowballValue(int weight, int time, int quality) {
    // Calculate the value of a snowball using the given weight, time, and quality
    return pow(static_cast<double>(weight) / time, quality);
}

int main() {
    int n;
    std::cin >> n;  // Read the number of snowballs

    double highestValue = -1;  // Initialize with a value lower than any possible snowball value
    int bestWeight, bestTime, bestQuality;

    // Iterate through each snowball
    for (int i = 0; i < n; ++i) {
        int weight, time, quality;
        std::cin >> weight >> time >> quality;

        // Calculate the value of the current snowball
        double snowballValue = calculateSnowballValue(weight, time, quality);

        // Update the highest value if necessary
        if (snowballValue > highestValue) {
            highestValue = snowballValue;
            bestWeight = weight;
            bestTime = time;
            bestQuality = quality;
        }
    }

    // Print the details of the snowball with the highest value
    std::cout << bestWeight << " : " << bestTime << " = " << static_cast<int>(highestValue + 0.5) << " (" << bestQuality << ")\n";

    return 0;
}

File: Snowballs.py
def calculate_snowball_value(weight, time, quality):
    """
    Calculate the value of a snowball using the given weight, time, and quality.

    Args:
        weight (int): The weight of the snowball.
        time (int): The time needed for the snowball to get to its target.
        quality (int): The quality of the snowball.

    Returns:
        float: The calculated snowball value.
    """
    return (weight / time) ** quality


def main():
    # Read the number of snowballs
    n = int(input())

    highest_value = -1  # Initialize with a value lower than any possible snowball value

    # Iterate through each snowball
    for _ in range(n):
        weight = int(input())
        time = int(input())
        quality = int(input())

        # Calculate the value of the current snowball
        snowball_value = calculate_snowball_value(weight, time, quality)

        # Update the highest value if necessary
        if snowball_value > highest_value:
            highest_value = snowball_value
            best_weight = weight
            best_time = time
            best_quality = quality

    # Print the details of the snowball with the highest value
    print(f"{best_weight} : {best_time} = {highest_value:.0f} ({best_quality})")


if __name__ == "__main__":
    main()
